### PR TITLE
Tests for Teddy compiling via string and file

### PR DESCRIPTION
**Pull request by alallier**
This PR serves as both a question/discussion and possible enhancement.

`teddy.compile(pathToFile)` seems to work as I expect, but `teddy.compile(someString)` does not.

According to the API it is suppose to populate `teddy.templates`, the compiling via file does populate `teddy.templates` but the string doesn't. But this also poses a question, what do we use for a key when we are compiling a string template. I see it is suppose to use [`name` which is equal to `template`](https://github.com/rooseveltframework/teddy/blob/b6da8210f0073f2eccb4db785bda1c98334f0913/teddy.js#L198-L199), this works in the filename case but doesn't make sense in the string case.

Let me know if I am approaching this incorrectly or if I can clarify further.

### CODE REVIEW

**Comment by Autre31415** [comment]
Relevant issue: #134

**Comment by alallier** [comment]
So it's a known issue that Teddy is broken when compiling via a string?

File: test/misc.js [hunk 1]
```diff
@@ -24,6 +24,21 @@ describe('Misc', function () {
     }
   })
 
+  it('should compile a template with source code as the parameter and populate teddy.templates', function (done) {
+    var templateAsString = '{! should properly escape HTML entities present in {variables} !} <p>{escapeTest}</p>'
+    var templateNameAndPath = 'something'
```
Comment (alallier) [inline test/misc.js]: Obviously don't merge with this

Comment (alallier) [inline test/misc.js]: Obviously don't merge with this

**Comment by Autre31415** [comment]
Strings definitely work, as it's a requirement for client side teddy use. Also, every template is compiled as a string here https://github.com/rooseveltframework/teddy/blob/master/test/client.js#L11

**Comment by kethinov** [comment]
Right now the Node.js context and the client-side context differ in how you can compile/render templates.

In Node.js the `fs` module exists, so you can do look ups based on file name. But client-side there is no way to look up anything by file name, so you have to manually name things. Issue #134 is about making this easier to do by adding a supported way of doing it via `compile`, but it can still be done today by directly addressing teddy's global template cache:

So, Node.js you can do this:

`teddy.compile('something')` this will locate the `something` file, open its code, compile the template, and name the compiled template 'something'.

But in the client context, it's a more manual process:

`teddy.templates['desiredName'] = teddy.compile(code)`

**Comment by alallier** [comment]
Thanks for the clarification, we might want to add a similar clarification to the README as well

**Comment by kethinov** [comment]
It's there: https://github.com/rooseveltframework/teddy#api-documentation

The problem isn't the docs, it's the API itself. It's primitive and hard to follow, thus #134.

**Comment by alallier** [comment]
It's this line that confuses me 

> Populates teddy.templates with the new template in the format of teddy.templates[path]: compiledSource. 

as it won't do that if you just provide a string

**Comment by kethinov** [comment]
Yeah, as I said, it's hard to follow. But it does explicitly say pathing is only available in the Node.js context. You can't expect it to index the templates object by path when path is impossible to discern client-side. As such, client-side compiles are indexed either by source IIRC.

**Comment by kethinov** [comment]
That is, unless you manually create an entry in `teddy.templates`. #134 is about automating that and discouraging manually messing with `teddy.templates`.

**Comment by alallier** [comment]
Ah I see the parenthesis now that you point it out. And yes exactly why I was getting confused. Alright thanks for clearing it up
